etl-pipeline/tests/integration/test_grin_client.py

```diff
@@ -0,0 +1,18 @@
+import pytest
+
+
+@pytest.mark.parametrize(
+    "fragment", ["_available", "_in_process", "_converted", "_failed", "_all_books"]
+)
+class TestGRINClient:
+    def test_url(self, grin_client, fragment):
```

Comment:
i think if you want we could bump this to a functional or unit test and focus these tests on just ensuring the requests to GRIN are solid.